### ISSUE TITLE
WinRM password

### ISSUE THREAD

**Issue by DaveBlooman** [mitchellh/packer#3550]
I need to build an AMI manually for various reasons, I then want to bake a new AMI based on this initial AMI using Packer.

I am using the Windows 2012 R2 AMI from the AWS market, which I can get a password for via the AWS console.  I use this password for the WinRM password in my Packer template, but it hangs at `Waiting for WinRM to become available`, I can remote into this instance however using this password.

Is there anything I need to do before I snapshot my initial AMI to make Packer and WinRM work, or set a specific password for WinRM?

**Comment by cbednarski** [comment]
The mailing list is a better place to ask open-ended questions like these since you can have some discussion about it with other users who do these types of builds. I don't do a lot of windows builds so I'm not sure what to suggest here. Some general things off the top of my head:

- Try running with `PACKER_LOG=1` set in your environment. [Instructions here.](https://www.packer.io/docs/other/debugging.html)
- "Waiting on ..." usually means one of two things: Either packer can't reach the instance because of a networking issue (firewall, security group, service is running on the wrong port, etc.) or the service Packer is waiting on never starts up. Setting `PACKER_LOG` will probably show you what is going and help you assess which of these is happening.

**Comment by ricardclau** [comment]
Hi @DaveBlooman 

Many chances are that something is blocking port 5985 to your instance. I understand you are able to RDP on port 3389, right?

What we are doing with our windows builds is:

- Build the instance in a security group which has port 5985 open from the packer build machine
- Bootstrap the instance supplying some userdata that ensures port 5985 is also open in the Windows firewall
- As a bonus, it is usually important to elevate powershell permissions

So, you would add in packer something like:

```
"user_data_file": "path_to_some_initial_config.txt",
```
And this would be the contents of the referred file (please adapt it to fit your particular needs):

```
<powershell>
# turn off PowerShell execution policy restrictions
Set-ExecutionPolicy -ExecutionPolicy Bypass -Scope LocalMachine

# configure WinRM
winrm quickconfig -q
winrm set winrm/config/winrs '@{MaxMemoryPerShellMB="2048"}'
winrm set winrm/config/winrs '@{MaxConcurrentUsers="100"}'
winrm set winrm/config/winrs '@{MaxProcessesPerShell="0"}'
winrm set winrm/config/winrs '@{MaxShellsPerUser="0"}'
winrm set winrm/config '@{MaxTimeoutms="7200000"}'
winrm set winrm/config/service '@{AllowUnencrypted="true"}'
winrm set winrm/config/service/auth '@{Basic="true"}'
winrm set winrm/config/service/auth '@{CredSSP="true"}'
winrm set winrm/config/client '@{TrustedHosts="*"}'


# open port 5985 in the internal Windows firewall to allow WinRM communication
netsh advfirewall firewall add rule name="WinRM 5985" protocol=TCP dir=in localport=5985 action=allow

net stop winrm
sc config winrm start=auto
net start winrm

</powershell>
```

Also, remember to "sysprep" your instance allowing hostname changes, etc before packer stops modifying the ec2config xml files

You can see the full thing in a small github repo I created for a talk I did in a local Winops meetup at https://github.com/ricardclau/packer-demos/tree/master/templates/windows-server-2012

Hope this helps!

**Comment by mwhooker** [comment]
Closing since it looks like this has been answered. If you're still having issues, I think the best way to get help is to mail the mailing list. Otherwise, if you think there's a bug in packer, feel free to open another issue. Thanks
